workbench-core/accounts/src/services/projectService.ts

```diff
@@ -127,91 +141,58 @@ export default class ProjectService {
+        projectsOnPage = manualSortProjects(sort, projects);
+      } else if (filter === undefined) {
+        // default sort is by createdAt
+        projectsOnPage = manualSortProjects({ createdAt: 'asc' }, projects);
       }
-
-      // juuuust the right amount--getPaginationToken from response and exit loop
-      else if (returnedCount === pageSize) {
-        paginationToken = getPaginationToken(projectsResponse);
-        projects = projects.concat(projectsResponse.Items.map((item) => this._formatFromDDB(item)));
+      // build page and pagination token
+      if (paginationToken) {
+        const manualExclusiveStartKey = fromPaginationToken(paginationToken);
+        const exclusiveStartProjectId = manualExclusiveStartKey.pk.split('#')[1];
+        const exclusiveStartProject = projectsOnPage.find(
+          (project) => project.id === exclusiveStartProjectId
+        );
+        if (exclusiveStartProject === undefined) {
+          throw Boom.badImplementation('Something went wrong with listing projects');
+        }
+        const indexOfExclusiveStartProject = projectsOnPage.indexOf(exclusiveStartProject);
+        projectsOnPage = projectsOnPage.slice(indexOfExclusiveStartProject + 1);
       }
-
-      // too much--take enough to fill and manually get pagination token from reponse and exit loop
-      else if (returnedCount > pageSize) {
-        const leftoverAmount = returnedCount - pageSize;
-        const totalProjectsFromResponse = projectsResponse.Items.map((item) => this._formatFromDDB(item));
-        projects = projects.concat(totalProjectsFromResponse.slice(0, -leftoverAmount));
-        const manualLastEvaluatedItem =
-          projectsResponse.Items[projectsResponse.Items.length - (leftoverAmount + 1)]; // this will be the whole entry
-        // If we query on GSI, then the LastEvaluatedKey will be the compose of GSI partition key, GSI sort key, primary partition key and primary sort key.
-        // TODO--extend to other GSIs after filtering is added
-        const manualLastEvaluatedKey = {
-          pk: manualLastEvaluatedItem.pk,
-          sk: manualLastEvaluatedItem.sk,
-          resourceType: manualLastEvaluatedItem.resourceType,
-          createdAt: manualLastEvaluatedItem.createdAt
+      if (projectsOnPage.length < pageSize) {
+        return {
+          data: projectsOnPage,
+          paginationToken: undefined
         };
-        paginationToken = Buffer.from(JSON.stringify(manualLastEvaluatedKey)).toString('base64');
-      }
-      if (paginationToken === undefined) {
-        break;
+      } else {
+        projectsOnPage = projectsOnPage.slice(0, pageSize);
+        const manualLastEvaluatedKey = buildDynamoDBPkSk(
+          projectsOnPage[pageSize - 1].id,
+          resourceTypeToKey.project
+        );
+        return { data: projectsOnPage, paginationToken: toPaginationToken(manualLastEvaluatedKey) };
       }
+    } else {
+      return { data: [], paginationToken: undefined };
     }
```

Comment:
It was a little confusing getting here and going back to see what the if was for this else. Might improve readability if we early return if `projectsResponse.Responses` is falsy

Reply:
so switch the `if` condition in line 152 to be a `!projectsResponse.Responses`? How would you feel about a comment in the else block says `// no responses returned from DDB`. I feel like having the `if` being the negative case isn't normal.

Reply:
I agree with Tyler. If it helps, the way I think of early returns is "if this case happens, dont worry about continuing on, end here"